test/util_midgard.cc

```diff
@@ -378,6 +378,34 @@ TEST(UtilMidgard, TestSimilarAndEqual) {
       << "Similar test fails for values with opposing signs";
 }
 
+// Fixes case in https://github.com/valhalla/valhalla/issues/2201#issuecomment-582656499
+TEST(UtilMidgard, TrimShapeAsan) {
+  float start = 42.2698097;
+  float end = 47;
+  PointLL start_vertex{8.54709148, 47.3651924};
+  PointLL end_vertex{8.54711914, 47.3651543};
+  std::vector<PointLL> shape = {
+      PointLL{8.5468483, 47.3655319},
+      PointLL{8.54691314, 47.365448},
+      PointLL{8.54711914, 47.3651543},
+  };
+  TrimShape(start, start_vertex, end, end_vertex, shape);
```

Comment:
woudlnt this illustrate the ASAN fix better if you used `shape.front()` and `shape.back()`?

Reply:
Fixed